fix(STONEINTG-598): clarify required tests (#234)
redhat-appstudio/docs.appstudio.io: docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_adding_an_integration_test.adoc

= Adding an integration test

> You can add an integration test to ensure that individual components correctly integrate to form a whole application. {ProductName} runs integration tests on the container images of components before it deploys those images.

Reviewer: ```suggestion
in {ProductName}, you can add integration tests to verify that the individual components of your application integrate correctly, forming a complete and functional application. {ProductName} runs these integration tests on the container images of components before their deployment.
```

Author: actually I didn't change this, I only accidentally pushed whitespace removal done by IDE. But can include it, no problem

. Optional: If you need to deploy your application as part of the test, use the drop-down menu in the *Environment* field to select *development*. {ProductName} then deploys your application to a temporary clone of that environment during testing.
+
WARNING: Without specifying an Environment, an integration test can only do static testing. To make application endpoints available for testing, you must deploy your application to an ephemeral environment.

> . Optional: *Mark as optional for release* if you want to allow this test scenario to fail without affecting deployment and release of the application.

Reviewer: ```suggestion
. Optional: You can choose to select *Mark as optional for release*. Doing so allows the integration tests to fail without impacting the deployment and release process of your application. 
```

+

> NOTE: By default all integration test scenarios are required to pass. A failing integration test marks the application snapshot as failed and prevents deploying and releasing of the application. By marking integration test scenario as optional this test will not prevent the application snapshot to be deployed and released even if it fails.

Reviewer 2: LGTM - some useful context here

> NOTE: By default all integration test scenarios are required to pass. A failing integration test marks the application snapshot as failed and prevents deploying and releasing of the application. By marking integration test scenario as optional this test will not prevent the application snapshot to be deployed and released even if it fails.

Reviewer: ```suggestion
NOTE: By default, all integration test scenarios are mandatory and must pass. A failing integration test marks the application snapshot as failed, preventing its deployment and release. However,  if you have selected *Mark as optional for release*, a failure in this test won't hinder the deployment and release of the application snapshot.
```
